packages/api/db/migration/20241016204145_create_animal_movement_task_tables.js

```diff
@@ -0,0 +1,112 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+export const up = async (knex) => {
+  // Add new animal_movement entry to task_type table
+  await knex('task_type').insert({
+    task_name: 'Animal Movement',
+    task_translation_key: 'ANIMAL_MOVEMENT',
+  });
+
+  // Add location_id column to animal and animal_batch tables (foreign key to location table)
+  await knex.schema.alterTable('animal', (table) => {
+    table.uuid('location_id').references('location_id').inTable('location');
+  });
+  await knex.schema.alterTable('animal_batch', (table) => {
+    table.uuid('location_id').references('location_id').inTable('location');
+  });
+
+  // Create animal_movement_purpose enum table
+  await knex.schema.createTable('animal_movement_purpose', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+  await knex('animal_movement_purpose').insert(
+    [
+      'GRAZING',
+      'MILKING',
+      'MANURE_MANAGEMENT',
+      'PEST_OR_DISEASE_CONTROL',
+      'BREEDING',
+      'FEEDING',
+      'VETERINARY_CARE',
+      'BIRTHING',
+      'WEANING',
+      'TRANSPORTATION',
+      'BEHAVIORAL_AND_SOCIAL_MANAGEMENT',
+      'EXERCISE',
+      'QUARANTINE',
+      'OTHER',
+    ].map((key) => ({ key })),
+  );
+
+  // Create animal_movement_task table
+  await knex.schema.createTable('animal_movement_task', (table) => {
+    table.integer('task_id').references('task_id').inTable('task').primary();
```

Comment:
should this table have a `toLocation` ?

Reply:
Previously I thought it should (and I added), but I realized that we're using `location_tasks` as "toLocation". Do you think it'll work?

Reply:
yes good ok that makes sense! Thanks for clarifying!